### ISSUE TITLE
[java] Merge rules about protected in final class

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#2974]
I think we should merge [AvoidProtectedFieldInFinalClass](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#avoidprotectedfieldinfinalclass) and [AvoidProtectedMethodInFinalClassNotExtending](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#avoidprotectedmethodinfinalclassnotextending). Just like for #2892, it's useless to have a distinction when the fix is the same. The names are also long and clumsy, and give too much detail about the implementation, which doesn't allow us to extend the behavior of the rules.

Proposed rule name: "ProtectedMemberInFinalClass" in design.xml and not codestyle.

Other improvements we can make to this new rule:
* protected nested types
* protected members in an anonymous class, enum, or record
* only ignore methods that override an inherited method (so stop checking blindly for an ExtendsList)

For example
```java
final class MyList extends AbstractList<String> {

    // a false negative of the current rule
    protected void idontOverrideAnything() { }

}
```
